cpp/nvtx3.hpp

```diff
@@ -502,22 +548,39 @@
  *
  */
 
-/**
- * @brief Enables the use of constexpr when support for C++14 relaxed constexpr
+/* Temporary helper #defines, removed with #undef at end of header */
+
+/* Within this header, nvtx3::NVTX3_VERSION_NAMESPACE resolves to nvtx3::vX,
+ * where "X" is the major version number. */
+#define NVTX3_CONCAT(A, B) A##B
+#define NVTX3_NAMESPACE_FOR(VERSION) NVTX3_CONCAT(v, VERSION)
+#define NVTX3_VERSION_NAMESPACE NVTX3_NAMESPACE_FOR(NVTX3_CPP_VERSION_MAJOR)
```

Comment:
Seems cleaner to put the `inline` check here instead of below where the namespace is actually opened. 
```suggestion
#define NVTX3_VERSION_NAMESPACE NVTX3_NAMESPACE_FOR(NVTX3_CPP_VERSION_MAJOR)
#if defined(NVTX3_INLINE_THIS_VERSION)
#define NVTX3_NAMESPACE inline NVTX3_VERSION_NAMESPACE
#else
#define NVTX3_NAMESPACE NVTX3_VERSION_NAMESPACE
#endif
```

Reply:
Interesting idea.  I was just following pattern I saw in a few places, like here:

https://stackoverflow.com/questions/11016220/what-are-inline-namespaces-for

I'd have to do it a little differently than you suggested because your macros here resolve to `namespace inline v1` and it needs to be `inline namespace v1`.  So we could do
```
#if defined(NVTX3_INLINE_THIS_VERSION)
#define NVTX3_INLINE_IF_REQUESTED inline
#else
#define NVTX3_INLINE_IF_REQUESTED
#endif
```
 then the usage below would look like `NVTX3_INLINE_IF_REQUESTED namespace NVTX3_VERSION_NAMESPACE {`.  Do you like that better than the way I currently have it?

Reply:
Note also that the `#if defined(NVTX3_INLINE_THIS_VERSION)` thing is still needed below in each minor version where it defines the NVTX3_WHATEVER macros as aliases of NVTX3_V1_WHATEVER, so this change doesn't let us be totally free of that pattern.

Reply:
OK I made this change and I liked it better than what I had.  Fixed in both the V1_0 impl and the example code in comments describing how to add new functionality in minor versions.